source/pervasive/seq_lib.rs

```diff
@@ -31,14 +33,101 @@ impl<A> Seq<A> {
     /// thereby 1 smaller.
     ///
     /// If the input sequence is empty, the result is meaningless and arbitrary.
-
     pub open spec fn drop_last(self) -> Seq<A>
         recommends self.len() >= 1
     {
         self.subrange(0, self.len() as int - 1)
-    } 
+    }
+
+    /// returns `true` if the sequequence has no duplicate elements
+    pub open spec fn no_duplicates(self) -> bool {
+        forall(|i, j| (0 <= i < self.len() && 0 <= j < self.len() && i != j)
+            ==> self[i] != self[j])
+    }
+
+    /// Returns `true` if two sequences are disjoint
+    pub open spec fn disjoint(self, other: Self) -> bool {
+        forall |a: A| self.contains(a) ==> !other.contains(a)
```

Comment:
This has three quantifiers and it has alternating forall/exists quantifiers.  It would be better to write it with just one quantifier:

```
forall|i: int, j: int| 0 <= i < self.len() && 0 <= j < other.len() ==> self[i] != other[j]
```

Reply:
Replaced definition with suggested forall in second commit.